src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -65,12 +66,16 @@ private enum Flags
                 _backingFieldFlags |= Flags.IsCreatedForFieldKeyword;
             }
 
-
             if (isEarlyConstructed)
             {
                 _backingFieldFlags |= Flags.IsEarlyConstructed;
             }
 
+            if (ContainingType.IsInterface && !isStatic)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterfacesCantContainFields, ErrorLocation);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_InterfacesCantContainFields, ErrorLocation); [](http://example.com/codeflow?start=16&length=74)

It looks like this approach comes with several shortcomings:
- location is not pointing to the syntax that is the source of the problem, which is the `field` keyword in the body.
- It looks like multiple uses of the `field` keyword will report the same diagnostics at the same location. Which will appear as a bunch of duplicate errors.
- There are scenarios when this symbol is created not in response to a `field` keyword. Those scenarios are already handled and reported. Reporting more diagnostics in those scenarios is not desired. In fact it looks like a lot of existing unrelated tests are now broken because of that.

I suggest trying to report the error in Binder_Expressions.cs instead. Next to this check:
```
                            if (IsInsideNameof)
                            {
                                Error(diagnostics, ErrorCode.ERR_FieldKeywordInsideNameOf, node);
                            }
```

And report it with location of the `field` keyword.